test: Hibernate jsonb support (#558)
GoogleCloudPlatform/pgadapter: samples/java/hibernate/README.md

| timestamptz / timestamp with time zone | LocalDateTime     |
| bytea                                  | byte[]            |
| date                                   | LocalDate         |
| jsonb                                  | Custom Data Type  |

### JSONB

> Hibernate 5.* doest not have native support for JSONB. For this, custom data type

Reviewer: nit:
```suggestion
Hibernate 5.* does not have native support for JSONB. For this, a custom data type
```

Author: Done.

will have to be added. There are modules available that provide support for this.
For the sample, we have used [vladmihalcea/hibernate-types](https://github.com/vladmihalcea/hibernate-types).

## Limitations
The following limitations are currently known:

| Limitation                 | Workaround                                                                                                                                                                                                                                                                   |
|----------------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|

> | Hibernate Version          | Currently, we only support 5.3.x.                                                                                                                                                                                                                                            |

Reviewer: I think we should make this a little simpler / less suggestive (the current text sounds like we're actively working on other versions)

```suggestion
| Hibernate Version          | Version 5.3.x. is supported                                                                                                                                                                                                                                           |
```

Do we know that it breaks with other versions? Or is this just a safeguard?

Author: For hibernate, this is just a safe guard but for Postgresql driver we know that it doesn't work with v42.5.0.

Reviewer: Would you mind sharing (offline) the SQL statement that 42.5.0 generates that fails?

> | PostgreSql Dialect Version | Currently, we only support 42.4.3.                                                                                                                                                                                                                                           |

Reviewer: Should this be:
```suggestion
| PostgreSql JDBC driver Version | Currently, we only support 42.4.3.                                                                                                                                                                                                                                           |
```
?

Author: Done
